generator/sources/java/src/com/kaltura/client/KalturaClientBase.java

```diff
@@ -88,7 +90,7 @@
  *
  */
 @SuppressWarnings("serial")
-abstract public class KalturaClientBase implements Serializable {
+abstract public class KalturaClientBase extends ReentrantLock implements Serializable {
```

Comment:
I think it's a better practice that KalturaClientBase will own a lock, rather than be a lock.
You can declare a field:
protected/private ReentrantLock lock = new ReentrantLock();